src/main/java/com/autotune/analyzer/experiment/ExperimentInitiator.java

```diff
@@ -92,5 +95,35 @@ public void validateAndUpdateResults(
             validationOutputData.setMessage("Exception occurred while validating the result data: " + e.getMessage());
             validationOutputData.setErrorCode(HttpServletResponse.SC_INTERNAL_SERVER_ERROR);
         }
+        return validationOutputData;
+    }
+
+    // Generate recommendations and add it to the kruize object
+    public boolean generateAndAddRecommendations(Map<String, KruizeObject> experimentsMap, List<String> experimentNameList) {
+        if (null == experimentNameList)
+            return false;
+        if (experimentNameList.size() == 0)
+            return false;
+        for (String experimentName: experimentNameList) {
+            if (!experimentsMap.containsKey(experimentName))
```

Comment:
In the current case, is experimentNameList consisting of a single element in the array? If so this is ok for now. If not we cannot bail out just because there is one bad element. We need to capture the bad element and return the error. Please mark a TODO here, so that we can come back to it while handling bulk API